doc/overview.md

```diff
@@ -0,0 +1,79 @@
+# Crucible Overview
+
+References to Crucible modules should all start with `Lang.Crucible`, which
+we omit for brevity.
+
+## Types
+
+The types of the Crucible language are defined in `Types`.
+  Of interest:
+    * `CrucibleType` is the list of types
+    * `TypeRepr` is the datatype of types, used when we want to store or pass
+      around Crucible types.
+
+
+## Values
+
+The inhabitants of each type are specified via the type function `RegValue`
+(`Simulator.RegValue`).  We also have `RegValue'` which is just
+a newtype wrapper `RegValue`, so that we can partially apply it.
```

Comment:
I'd imagine that this sentence would be a bit opaque to those not already familiar with `-XTypeFamilies`. I'm not sure exactly what the best solution is, though, because I can't find the prohibition on partial application mentioned anywhere in [the official docs](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/type_families.html).